compiler/rustc_typeck/src/check/fn_ctxt/_impl.rs

```diff
@@ -1061,11 +1057,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     last_def_id, last_bounds, exp_def_id, exp_bounds
                 );
 
-                let (last_local_id, exp_local_id) =
-                    match (last_def_id.as_local(), exp_def_id.as_local()) {
-                        (Some(last_hir_id), Some(exp_hir_id)) => (last_hir_id, exp_hir_id),
-                        (_, _) => return None,
-                    };
+                let (Some(last_local_id), Some(exp_local_id)) = (last_def_id.as_local(), exp_def_id.as_local()) else {
```

Comment:
This should probably just be two separate assignments using the question mark operator.